packages/foundation-models/README.md

```diff
@@ -145,6 +147,82 @@ logger.info(
 
 Refer to `AzureOpenAiChatCompletionParameters` interface for other parameters that can be passed to the chat completion request.
 
+#### Streaming
+
+The `AzureOpenAiChatClient` supports streaming response for chat completion requests based on the [Server-sent events](https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events) standard.
+
+Use the `stream()` method to receive a stream of chunk responses from the model.
+After consuming the stream, call the helper methods to get the finish reason and token usage information respectively.
+
+```ts
+const chatClient = new AzureOpenAiChatClient('gpt-4o');
+const response = await chatClient.stream({
+  messages: [
+    {
+      role: 'user',
+      content: 'Give me a very long introduction of SAP Cloud SDK.'
+    }
+  ]
+});
+
+for await (const chunk of response.stream) {
+  console.log(JSON.stringify(chunk));
+}
+
+const finishReason = response.getFinishReason();
+const tokenUsage = response.getTokenUsage();
+
+console.log(`Finish reason: ${finishReason}\n`);
+console.log(`Token usage: ${JSON.stringify(tokenUsage)}\n`);
+```
+
+##### Streaming the Delta Content
+
+The client provides a helper method to extract delta content and stream string directly.
```

Comment:
[suggestion] I wouldn't understand that the delta content contains the actual LLM output we are interested in, without knowing the structure with which OpenAI responds.

Should we consider also giving an example of how a data chunk looks like?

Reply:
Normally, `data` should be typed. This will be fixed once we update the client with `2024-10-21` spec.